src/mimeparser.rs

```diff
@@ -3820,4 +3820,25 @@ Content-Disposition: reaction\n\
 
         Ok(())
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_long_and_duplicated_filenames() -> Result<()> {
+        let mut tcm = TestContextManager::new();
+        let alice = tcm.alice().await;
+        let bob = tcm.bob().await;
+
+        bob.set_config(Config::Displayname, Some("bob")).await?;
```

Comment:
Maybe better not just "bob", but smth like "Bob Smith". "bob" can be guessed from the email address